### PR TITLE
Don't reexecute logout command when session is expired

### PR DESCRIPTION
**Pull request by rkovarik**
… infinite loop

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -56,5 +69,11 @@
       <version>1.9.5</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (larrychu-ibm) [inline pom.xml]: We use testNG for unit tests, not junit. Please change the test to a TestNG test.

File: src/test/java/com/silverpop/api/client/xmlapi/XmlApiClientTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import com.silverpop.api.client.xmlapi.util.XmlApiClientFactory;
+
+public class XmlApiClientTest {
+
+    private XmlApiClient client = null;
+
+    @Test(expected = ApiResultException.class)
+    public void logoutWhenSessionIsExpired() throws ApiResultException {
+        //GIVEN
+        XmlApiClientFactory factory = new XmlApiClientFactory() {
```
Comment (larrychu-ibm) [inline src/test/java/com/silverpop/api/client/xmlapi/XmlApiClientTest.java]: Please use Mockito to automatically generate a test double and use Mockito.when to stub a method to provide a return value.

File: src/test/java/com/silverpop/api/client/xmlapi/XmlApiClientTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        XmlApiSession session = new XmlApiSession("", new LoginCommand(), factory) {
+            @Override
+            public boolean isOpen() {
+                return true;
+            }
+        };
+
+        client = new XmlApiClient(session) {
+            @Override
+            protected ApiResult validateSessionAndExecuteCommand(ApiCommand command, Map<String, String> requestHeaders) throws ApiResultException {
```
Comment (larrychu-ibm) [inline src/test/java/com/silverpop/api/client/xmlapi/XmlApiClientTest.java]: We prefer not to control test flow by overriding a protected method in the SUT. Please stub a collaborator to control flow when possible.

File: pom.xml [hunk 4]
```diff
@@ -17,6 +17,19 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
```
Comment (larrychu-ibm) [inline pom.xml]: Adding surefire plugin is out-of-scope of the issue that is addressed. Please remove.

**Comment by rkovarik** [comment]
Thx for review. I haven't found an easy way how to stub this so I just removed the test unless anybody is interested in rewriting the test...

**Pull request closed without merging**